AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -899,6 +890,51 @@ - (void)transitionContext:(_ASTransitionContext *)context didComplete:(BOOL)didC
   _transitionContext = nil;
 }
 
+#pragma mark - Layout
+
+- (void)_completePendingLayoutTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_pendingLayoutTransition) {
+    [self _setCalculatedLayout:_pendingLayoutTransition.pendingLayout];
+    [self _completeTransition:_pendingLayoutTransition];
+    _pendingLayoutTransition = nil;
+  }
+  [self _completeLayoutCalculation];
+}
+
+- (void)_setCalculatedLayout:(ASLayout *)layout
+{
+  ASDN::MutexLocker l(_propertyLock);
+  
+  ASDisplayNodeAssertTrue(layout.layoutableObject == self);
+  ASDisplayNodeAssertTrue(layout.size.width >= 0.0);
+  ASDisplayNodeAssertTrue(layout.size.height >= 0.0);
+  
+  _layout = layout;
+}
+
+- (void)_completeTransition:(ASLayoutTransition *)layoutTransition
+{
+  // Layout transition is not supported for non implicit hierarchy managed nodes
```

Comment:
Unfortunately we cannot support it at the moment. This is one of the two problems we need to look into in the current implementation of implicit hierarchy management / layout transitions. Why we currently cannot support that is that we will run into an assert: A flattened layout must consist exclusively of node sublayouts.

What actually can happen is that in the previousLayout that is referenced in a layout transition the layoutable objects in the sublayouts can become nil. As the layoutable is weak in the layout it can happen that they are deallocated before the actual transition can finish. You can reproduce it very well in the ASDKGram example if you remove the `self.usesImplicitHierarchyManagement == NO` in this code path, rebuild, and just scroll down the feed. The reason in this particular example is that the we remove the Comment Nodes (ASTextNodes) from the supernodes and after that call ‘setNeedsLayout’. This has the effect that the Comment Nodes will be deallocated and become nil in the sublayouts of `pendingLayout` we use for the layout transition.

The solution would be to have some kind of managed nodes (ihm managed) / unmanaged notes (manual managed nodes) and dirty / undirty layout, so we can use this information in the layout transition to go from one hierarchy to another.